Backpack/src/androidTest/java/net/skyscanner/backpack/calendar2/data/CalendarRangeSelectionTests.kt

```diff
@@ -35,65 +36,69 @@ import org.junit.Test
 class CalendarRangeSelectionTests {
 
   private val rangeSelection = CalendarSettings.Default.copy(
-    selectionMode = CalendarParams.SelectionMode.Range,
+    selectionMode = CalendarParams.SelectionMode.Dates,
+  )
+
+  private val monthSelection = CalendarSettings.Default.copy(
+    selectionMode = CalendarParams.SelectionMode.Month("Select whole month"),
   )
 
   @Test
   fun when_range_is_opened_selection_is_correct() {
     testCalendarWith(rangeSelection) {
-      stateMachine.onClick(firstDay)
+      stateMachine.onClick(CalendarAction.CalendarDayAction(firstDay))
 
       verify {
-        assertEquals(CalendarSelection.Range(start = firstDay.date, end = null), state.selection)
+        assertEquals(CalendarSelection.Dates(start = firstDay.date, end = null), state.selection)
       }
     }
   }
 
   @Test
   fun when_range_is_closed_selection_is_correct() {
     testCalendarWith(rangeSelection) {
-      stateMachine.onClick(firstDay)
-      stateMachine.onClick(lastDay)
+      stateMachine.onClick(CalendarAction.CalendarDayAction(firstDay))
```

Comment:
I would still keep it as 2 methods instead of sealed classes here.

Reply:
@bvitaliyg Initially, I had these two methods, however, the introduction of CalendarAction improves readability as well as makes adding a new action/listener to the calendar in the future easier. Do you still want me to change it?

CC: @sergiomarqmoura & @marianeum, any objection  👆🏽 ?

Reply:
Yeah I also like that it makes the distinction between click events/user interactions and other state changes clearer, rather than a few functions that are connected by the name

Reply:
We can rename the methods to something like dispatchDayClicked(). Introducing Actions as a concept may not work with state machines in general, for instance – in some implementations Actions are used for another, SM-related comcepts.

Reply:
Or, we can rename it to something else rather than action – may interaction? CalendarInteraction.DateClicked, CalendarInteraction.SelectMonthClicked?